drivers/net/wireless/marvell/mwifiex/pcie.c

```diff
@@ -392,6 +393,13 @@ static void mwifiex_pcie_reset_prepare(struct pci_dev *pdev)
 	mwifiex_shutdown_sw(adapter);
 	clear_bit(MWIFIEX_IFACE_WORK_DEVICE_DUMP, &card->work_flags);
 	clear_bit(MWIFIEX_IFACE_WORK_CARD_RESET, &card->work_flags);
+
+	/* For Surface gen4+ devices, we need to put wifi into D3cold right
+	 * before performing FLR
+	 */
+	if (dmi_check_system(surface_gen4p_dmi_table))
```

Comment:
Quirks are usually implemented via bitflags, here that would be on the `card` I think, so the DMI matching is done once when the driver is loaded and based on that, bitflags are set. Then check those bitflags here and below instead of DMI matching.

Reply:
Got it. Addressed with commit https://github.com/linux-surface/kernel/pull/58/commits/206fa3cd1d885b4bb5d72b8b518a2b90d580d548